xla/service/gpu/tests/gpu_atomic_test.cc

```diff
@@ -78,6 +78,14 @@ CHECK-NOT: store atomic{{.*}}unordered, align 4
 }
 
 TEST_F(GpuAtomicTest, TestAddAtomicF32) {
+  if (is_built_with_rocm_ &&
```

Comment:
Let's create a separate test for AMD. I believe you should be able to use `GTEST_SKIP` in `SetUp` to prevent execution on Nvidia hardware then.